File: Front/src/app/models/Professeur.ts
export class Professeur {
  idProfesseur: number;
  idUser: number;
  nom: string;
  prenom: string;
  departement: string;
  groupes: string;

  constructor(
    idProfesseur: number,
    idUser: number,
    nom: string,
    prenom: string,
    departement: string,
    groupes: string
  ) {
    this.idProfesseur = idProfesseur;
    this.idUser = idUser;
    this.nom = nom;
    this.prenom = prenom;
    this.departement = departement;
    this.groupes = groupes;
  }
}

File: GestionAbsence-1/src/main/java/com/enicarthage/entities/Professeur.java
package com.enicarthage.entities;



import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Entity
@ToString
@EqualsAndHashCode

@Table(name = "professeurs")
public class Professeur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idProfesseur;

    
    private Long idUser;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private String prenom;

    @Column(nullable = false)
    private String departement;
    
    private String groupes;

	public Long getIdProfesseur() {
		return idProfesseur;
	}

	public void setIdProfesseur(Long idProfesseur) {
		this.idProfesseur = idProfesseur;
	}

	public Long getIdUser() {
		return idUser;
	}

	public void setIdUser(Long idUser) {
		this.idUser = idUser;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public String getDepartement() {
		return departement;
	}

	public void setDepartement(String departement) {
		this.departement = departement;
	}

	public String getGroupes() {
		return groupes;
	}

	public void setGroupes(String groupes) {
		this.groupes = groupes;
	}
    

    // Getters and Setters
}

